Add a fish-specific fix for #8633

WeetHet opened the pull request:
Release Notes:
- Fixed detection of `direnv` not working in `fish` when an LSP adapter (`gopls`, for example) tries to detect user-installed binaries. (#8633)

WeetHet committed:
Add a fish-specific fix for #8633

WeetHet commented:
@mrnugget could you review this?

WeetHet committed:
Simplify code a bit

mrnugget committed:
Simplify code a bit and update comment

mrnugget commented:
Great! I simplified code a bit and moved the comment around (and mentioned fish). Will merge once CI is green — you did test that this makes it work, right?

WeetHet commented:
> Great! I simplified code a bit and moved the comment around (and mentioned fish). Will merge once CI is green — you did test that this makes it work, right?

Yes, it works for me. I'd appreciate you double checking it though

WeetHet commented:
@mrnugget could you check if it works for you and if it does, merge this pr?
